frontend/src/views/TE/Upload.vue

```diff
@@ -190,6 +209,62 @@ export default class Upload extends Vue {
     }
   }
 
+  getFileHash(file: File) {
+    return new Promise(resolve => {
+      const fileReader = new FileReader();
+      fileReader.onload = function(event: any) {
+        const spark = new Spark.ArrayBuffer();
+        spark.append(event.target.result);
+        resolve(btoa(spark.end(true)));
+      };
+      fileReader.readAsArrayBuffer(file);
+    });
+  }
+
+  async validateFileHash(file: File, task: any) {
+    const fileHash = await this.getFileHash(file);
+    if (!task.versions) return;
+    for (const version of Object.values<any>(task.versions)) {
+      const ref = this.uploadsBucket.ref().child(`/${version.uploadPath}`);
+      const metadata = await ref.getMetadata().catch(e => "error");
+      if (metadata === "error") continue;
+      if (fileHash === metadata.md5Hash) {
+        throw new Error("You had uploaded the same file earlier.");
+      }
+    }
+  }
+
+  async validateUpload(file: File) {
+    validateFlacFile(file);
```

Comment:
```suggestion
    validateContentType(file);
```

Reply:
Moreover, I see that this function is validating both content type and file name.

I think due to our current refactoring, we can

- File name can be validated directly in this method as a step. Task ID regexp can be imported.
- Content type can be validated directly here as a step. It's just a line of code, and it doesn't worth having a separate function and tests for it.